config/config.go

```diff
@@ -32,3 +42,39 @@ func AddFlags(cmd *cobra.Command) {
 		panic(err)
 	}
 }
+
+func PersistentStartupProcessFlags() error {
+
+	if err := convertResolveHostsToMap(); err != nil {
+		return err
+	}
+	return nil
+
+}
+
+func convertResolveHostsToMap() error {
+	for _, resolveHost := range viper.GetStringSlice(optname.Resolve) {
+		split := strings.SplitN(resolveHost, ":", 3)
+		if len(split) != 3 {
+			return fmt.Errorf("invalid resolve host format, expected <hostname>:port:<ip>, got: %s", resolveHost)
+		}
+		host, port, addr := strings.ToLower(split[0]), split[1], split[2]
```

Comment:
I don't like the ToLower here, I don't think we need it

Reply:
I'd rather join `host` and `port` back into `addr` asap:
```suggestion
		addr, ipString := fmt.Sprint("%s:%s", strings.ToLower(split[0]), split[1]), split[2]
```
also it should be `addr` because that's what net.Dial calls it, so the other thing should be ip